mlir/utils/jenkins/Jenkinsfile.release

```diff
@@ -23,7 +23,7 @@ String dockerArgs() {
 }
 
 String dockerImage() {
-    return 'rocm/mlir:rocm5.1-latest'
```

Comment:
have been using rocm5.1 for release testing ? @jerryyin

Reply:
I don't think we have an active job for this `Jenkinsfile.release`. This job only caches the produced static library for reference, it has not been particularly useful.

The release branch testing job is using `Jenkinsfile` directly.